src/discussions/posts/post-actions-bar/PostActionsBar.jsx

```diff
@@ -26,11 +27,22 @@ function PostActionsBar({
   const dispatch = useDispatch();
   const loadingStatus = useSelector(selectconfigLoadingStatus);
   const blackoutDateRange = useSelector(selectBlackoutDate);
+  const hasPrivilege = useUserPrivilage();
 
   const handleCloseInContext = () => {
     postMessageToParent('learning.events.sidebar.close');
   };
 
+  const handleAddPost = () => {
+    if (loadingStatus === RequestStatus.SUCCESSFUL && (!(inBlackoutDateRange(blackoutDateRange)))) {
+      return true;
+    }
+    if (hasPrivilege) {
+      return true;
+    }
+    return false;
+  };
+
```

Comment:
You will not need redundant this check after the hook modification.  The hook will simply return a Boolean value.